src/main/java/htsjdk/samtools/util/zip/DeflaterFactory.java

```diff
@@ -43,36 +43,21 @@
  */
 public class DeflaterFactory {
 
-    private static Constructor<IntelDeflater> intelDeflaterConstructor;
+    private static final boolean isUsingIntelDeflater;
 
     static {
-        try {
-            if (Defaults.TRY_USE_INTEL_DEFLATER) {
-                final Class<IntelDeflater> clazz = (Class<IntelDeflater>) Class.forName("htsjdk.samtools.util.zip.IntelDeflater");
-                intelDeflaterConstructor = clazz.getConstructor(Integer.TYPE, Boolean.TYPE);
-            }
-        } catch (ClassNotFoundException e) {
-            intelDeflaterConstructor = null;
-        } catch (NoSuchMethodException e) {
-            intelDeflaterConstructor = null;
-        } catch (UnsatisfiedLinkError e) {
-            intelDeflaterConstructor = null;
-        }
+        isUsingIntelDeflater = Defaults.TRY_USE_INTEL_DEFLATER && (new IntelDeflater().load());
```

Comment:
I would like to keep the reflective lookup and instantiation in place since this allows users of htsjdk to just exclude the intel-gkl project when hooking in the htsjdk dependency and it will all still work.  Given this change users have no choice but to depend on this library and even if they chose to avoid using the custom deflator.